pep/app/pods/components/document/text/component.ts

```diff
@@ -76,23 +144,97 @@ export default class DocumentText extends Component<DocumentTextArgs> {
-            if (rx) {
-                const id = rx.split('.');
+            if (docId) {
+                const id = docId.split('.');
                 this.viewGlossaryTerm(target.innerText, id[id.length - 1]);
             } else if (groupName) {
                 this.viewGlossaryTerm(groupName);
             }
+        } else if (type === DocumentLinkTypes.BIBLIOGRAPHY) {
+            const id = attributes.getNamedItem('data-document-id')?.nodeValue;
+            if (id) {
+                this.router.transitionTo('read.document', id);
+            }
+        } else if (type === DocumentLinkTypes.DOCUMENT) {
+            const id = attributes.getNamedItem('data-document-id')?.nodeValue;
+            if (id) {
+                this.router.transitionTo('read.document', id);
+            }
+        } else if (type === DocumentLinkTypes.PAGE) {
+            const reference = attributes.getNamedItem('data-r')?.nodeValue;
+            const referenceArray = reference?.split(/\.(?=[^\.]+$)/) ?? [];
+            const documentId = referenceArray[0];
+            const apiPage = referenceArray[1];
+            const page = parseInt(apiPage.substring(1), 10);
+            if (documentId === this.args.document.id) {
+                //scroll to page number
+                this.scrollToPage(page);
+            } else {
+                //transition to a different document with a specific page
+                this.router.transitionTo('read.document', documentId, {
+                    queryParams: {
+                        page
+                    }
+                });
+            }
+        } else if (type === DocumentLinkTypes.FIGURE || type === DocumentLinkTypes.TABLE_FIGURE) {
+            const url = target?.getAttribute('src');
+            const parent = target.parentElement?.parentElement;
+            const caption = parent?.querySelector('.caption')?.innerHTML;
+            this.modal.open('document/image', {
+                url,
+                caption
+            });
+        } else if (type === DocumentLinkTypes.FIGURE_WITH_ID) {
+            const figureId = attributes.getNamedItem('data-figure-id')?.nodeValue;
+            const figure = this.containerElement?.querySelector(`#${figureId}`);
+            const image = figure?.querySelector('img');
+            const url = image?.getAttribute('src');
+            const caption = figure?.querySelector('.caption')?.innerHTML;
+            this.modal.open('document/image', {
+                url,
+                caption,
+                id: parseInt(figureId?.substring(1) ?? '', 10)
+            });
+        } else if (type === DocumentLinkTypes.AUTHOR_SEARCH) {
+            const firstName = target?.querySelector('.nfirst')?.innerHTML;
+            const lastName = target?.querySelector('.nlast')?.innerHTML;
+            const name = `${firstName} ${lastName}`;
+            this.args.viewSearch(JSON.stringify([{ term: name, type: SearchTermId.AUTHOR, value: name }]));
         }
```

Comment:
maybe not for this merge, but would be nice if we could break up this mammoth method, maybe with helper methods/functions for handing each of the different link types, and use a hash to map each type to its respective method, instead of a ginormous if/else block?

Reply:
I'll look into this when I have time. I've added a TODO